xbmc/windowing/win10/WinSystemWin10DX.cpp

```diff
@@ -153,3 +154,41 @@ void CWinSystemWin10DX::UninitHooks()
 void CWinSystemWin10DX::InitHooks(IDXGIOutput* pOutput)
 {
 }
+
+bool CWinSystemWin10DX::IsHDRDisplay()
+{
+  if (CWIN32Util::GetWindowsHDRStatus() != HDR_STATUS::HDR_UNSUPPORTED)
+    return true;
+
+  return false;
+}
+
+HDR_STATUS CWinSystemWin10DX::GetOSHDRStatus()
+{
+  return CWIN32Util::GetWindowsHDRStatus();
+}
+
+HDR_STATUS CWinSystemWin10DX::ToggleHDR()
+{
+  return m_deviceResources->ToggleHDR();
+}
+
+bool CWinSystemWin10DX::IsHDRSupported() const
+{
+  return m_deviceResources->IsHDRSupported();
```

Comment:
it looks like a copy of `CWinSystemWin10DX::IsHDRDisplay()` because `CDeviceResources` does the same check. isn't?

Reply:
`IsHDRSupported` is cached status of system supports HDR (Windows HDR switch is ON or OFF but is present) for use in `RenderBase::Configure` to auto switch HDR when is necessary.

There may be some negative performance impact if `GetOSHDRStatus` was used each time in `RenderBase::Configure`. Is not necessary because here is only need know that system supports HDR (not the current ON/OFF status)

```
  m_AutoSwitchHDR = DX::Windowing()->IsHDRSupported() &&
                    CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
                        DX::Windowing()->SETTING_WINSYSTEM_IS_HDR_DISPLAY);

  // Auto switch HDR only if supported and "Settings/Player/Use HDR display capabilities" = ON
  if (m_AutoSwitchHDR)
  {
....
```

Reply:
`RenderBase::Configure` is called rarely (at video starts and if video params are changed), I see no performance issue here.

Reply:
I think you're right and the impact here is minimal. I have changed IsHDRSupported by IsHDRDisplay and removed the leftover code. The change is in a new commit to make it easier to review, later I will put it together in one commit (squash).